src/AzureAuth/Commands/Ado/CommandToken.cs

```diff
@@ -3,59 +3,127 @@
     public class CommandToken
     {
+        private const string OutputOption = "--output";
+        private const string OutputOptionDescription = "How to print the token. One of [token, header, headervalue].\nDefault: token";
+
         /// <summary>
-        /// Gets or sets the Azure Tenant ID to use for authentication.
+        /// The available Token Formats.
         /// </summary>
+        public enum OutputMode
+        {
+            /// <summary>
+            /// Raw Token
+            /// </summary>
+            Token,
+
+            /// <summary>
+            /// Authorization http header value.
+            /// </summary>
+            Header,
+
+            /// <summary> TODO </summary>
+            HeaderValue,
+        }
+
+        [Option(OutputOption, OutputOptionDescription, CommandOptionType.SingleValue)]
+        private OutputMode Output { get; set; } = OutputMode.Token;
+
         [Option(CommandAad.TenantOption, Description = "The Azure Tenant ID to use for authentication. Defaults to Microsoft.")]
-        public string Tenant { get; set; } = AzureAuth.Ado.Constants.Tenant.Microsoft;
+        private string Tenant { get; set; } = AzureAuth.Ado.Constants.Tenant.Microsoft;
 
-        /// <summary>
-        /// Gets or sets the auth modes.
-        /// </summary>
         [Option(CommandAad.ModeOption, CommandAad.AuthModeHelperText, CommandOptionType.MultipleValue)]
-        public IEnumerable<AuthMode> AuthModes { get; set; } = new[] { AuthMode.Default };
+        private IEnumerable<AuthMode> AuthModes { get; set; } = new[] { AuthMode.Default };
 
-        /// <summary>
-        /// Gets or sets domain suffix to filter on.
-        /// </summary>
         [Option(CommandAad.DomainOption, Description = CommandAad.DomainHelpText)]
-        public string Domain { get; set; }
+        private string Domain { get; set; }
 
-        /// <summary>
-        /// Gets or sets the global timeout option.
-        /// </summary>
         [Option(CommandAad.TimeoutOption, CommandAad.TimeoutHelpText, CommandOptionType.SingleValue)]
-        public double Timeout { get; set; } = CommandAad.GlobalTimeout.TotalMinutes;
+        private double Timeout { get; set; } = CommandAad.GlobalTimeout.TotalMinutes;
+
+        [Option(CommandAad.PromptHintOption, CommandAad.PromptHintHelpText, CommandOptionType.SingleValue)]
+        private string PromptHint { get; set; }
 
         /// <summary>
-        /// Gets or sets the prompt hint.
+        /// Format a PAT based on <paramref name="output"/>.
         /// </summary>
-        [Option(CommandAad.PromptHintOption, CommandAad.PromptHintHelpText, CommandOptionType.SingleValue)]
-        public string PromptHint { get; set; }
+        /// <param name="value">The PAT value.</param>
+        /// <param name="output">The output mode.</param>
+        /// <param name="scheme">The <see cref="Authorization"/> scheme to use.</param>

```

Comment:
Scheme means basic or bearer?

Reply:
Correct, and doc comment has a link to the Authorization scheme Enum.